### PR TITLE
Apply clippy fixes for Rust 1.83

### PR DESCRIPTION
**Pull request by findepi**
`dev/rust_lint.sh` no longer passes for me, maybe because of `rustup update`. This is first portion of fixes suggested by clippy.

- fixes https://github.com/apache/datafusion/issues/13597

### CODE REVIEW

File: datafusion/expr-common/src/type_coercion/aggregates.rs [hunk 1]
```diff
@@ -294,19 +294,19 @@ pub fn coerce_avg_type(func_name: &str, arg_types: &[DataType]) -> Result<Vec<Da
     // Supported types smallint, int, bigint, real, double precision, decimal, or interval
     // Refer to https://www.postgresql.org/docs/8.2/functions-aggregate.html doc
     fn coerced_type(func_name: &str, data_type: &DataType) -> Result<DataType> {
-        return match &data_type {
+        match &data_type {
             DataType::Decimal128(p, s) => Ok(DataType::Decimal128(*p, *s)),
             DataType::Decimal256(p, s) => Ok(DataType::Decimal256(*p, *s)),
             d if d.is_numeric() => Ok(DataType::Float64),
-            DataType::Dictionary(_, v) => return coerced_type(func_name, v.as_ref()),
```
Comment (comphead) [inline datafusion/expr-common/src/type_coercion/aggregates.rs:301]: oh that becomes a clippy complaint finally

File: datafusion/expr/src/logical_plan/display.rs [hunk 2]
```diff
@@ -181,7 +181,7 @@ impl<'a, 'b> GraphvizVisitor<'a, 'b> {
     }
 }
 
-impl<'n, 'a, 'b> TreeNodeVisitor<'n> for GraphvizVisitor<'a, 'b> {
```
Comment (comphead) [inline datafusion/expr/src/logical_plan/display.rs:184]: the underscore does it mean unused lifetime?

File: datafusion/functions-nested/src/concat.rs [hunk 3]
```diff
@@ -438,7 +438,7 @@ fn concat_internal<O: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
     Ok(Arc::new(list_arr))
 }
 
-/// Kernal functions
+// Kernal functions
```
Comment (comphead) [inline datafusion/functions-nested/src/concat.rs:441]:
```suggestion
// Kernel functions
```

File: datafusion/optimizer/src/replace_distinct_aggregate.rs [hunk 4]
```diff
@@ -54,8 +54,6 @@ use datafusion_expr::{Aggregate, Distinct, DistinctOn, Expr, LogicalPlan};
 /// )
 /// ORDER BY a DESC
 /// ```
-
-/// Optimizer that replaces logical [[Distinct]] with a logical [[Aggregate]]
```
Comment (comphead) [inline datafusion/optimizer/src/replace_distinct_aggregate.rs:58]: do we need the comment to be removed?

File: datafusion/physical-plan/src/sorts/sort_preserving_merge.rs [hunk 5]
```diff
@@ -746,7 +746,7 @@ mod tests {
 
     // Split the provided record batch into multiple batch_size record batches
     fn split_batch(sorted: &RecordBatch, batch_size: usize) -> Vec<RecordBatch> {
-        let batches = (sorted.num_rows() + batch_size - 1) / batch_size;
```
Comment (comphead) [inline datafusion/physical-plan/src/sorts/sort_preserving_merge.rs:749]: was it a clippy complaint?

**Review by comphead — COMMENTED** [review]
Thanks @findepi 
Please help with the comments

File: datafusion/expr/src/logical_plan/display.rs (hunk 2 above)
Comment (Dandandan) [reply datafusion/expr/src/logical_plan/display.rs:184]: https://doc.rust-lang.org/reference/lifetime-elision.html

File: datafusion/optimizer/src/replace_distinct_aggregate.rs (hunk 4 above)
Comment (findepi) [reply datafusion/optimizer/src/replace_distinct_aggregate.rs:58]: it duplicates the first line of the long comment above, i don't think it's needed

File: datafusion/physical-plan/src/sorts/sort_preserving_merge.rs (hunk 5 above)
Comment (findepi) [reply datafusion/physical-plan/src/sorts/sort_preserving_merge.rs:749]: i did not modify this line manually. but i did run clippy with `--fix`, so yes

**Comment by findepi** [comment]
> I think we need to `#allow[missing_docs]` as the clippy now complains on that

:) just added

**Comment by findepi** [comment]
now that https://github.com/apache/datafusion/pull/13598 is merged, let be rebase, otherwise we no longer test with 1.83

**Comment by findepi** [comment]
[clippy](https://github.com/apache/datafusion/actions/runs/12075224493/job/33674907768?pr=13596#logs) job passed with toolchain pin dropped.

**Review by comphead — APPROVED** [review]
lgtm thanks @findepi

**Comment by jonahgao** [comment]
Thanks @findepi @comphead @Dandandan

**Pull request merged**